src/lib/forms/AutogenerateForm.svelte

```diff
@@ -0,0 +1,85 @@
+<script lang="ts">
+  import { slide } from "svelte/transition";
+  import type { Field } from "./types";
+
+  // This component creates a form to collect the property described by spec, and put the result in value
+  export let spec: Field;
+  export let value;
+
+  // Blank out initial values if needed
+  let oneOfCase = "";
+  if (typeof spec == "string") {
+  } else if ("members" in spec) {
```

Comment:
These're so much less clear than Rust's `if let` or `match`. I think I'll switch to https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates